Modules/Core/Common/include/itkShapedImageNeighborhoodRange.h

```diff
@@ -388,9 +388,13 @@ class ShapedImageNeighborhoodRange final
      */
     QualifiedIterator() = default;
 
-    /** Constructor that allows implicit conversion from non-const to const
-     * iterator. Also serves as copy-constructor of a non-const iterator.  */
-    QualifiedIterator(const QualifiedIterator<false> & arg) noexcept
+    QualifiedIterator(const QualifiedIterator<VIsConst> & arg) noexcept = default;
```

Comment:
Please consider the Rule of Zero here: No need to add an explicitly defaulted (`= default`) copy constructor. Instead, you may remove the existing explicitly defaulted assignment operator `operator=(const QualifiedIterator &)` as well.